### PR TITLE
Remove `Dummy` entity type

### PR DESCRIPTION
**Pull request by Alfred-Mountfield**
## 🌟 What is the purpose of this PR?

The Type System previously didn't allow for link destinations to be "anything". As such, we introduced a "Dummy" entity type as a stand in for that during HASH development, and this PR removes it now that the Type System allows you to leave the destinations unconstrained (expressed as an empty array).

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1201095311341924/1203449969023907/f) _(internal)_

## 🚫 Blocked by

N/A

## 🔍 What does this change?

- Removes usages of the dummy entity type

## 📜 Does this require a change to the docs?

No

## ⚠️ Known issues

None

## 🐾 Next steps

Update the type editors to allow for "any" link destinations.

## 🛡 What tests cover this?

Unsure about which specific ones, expect CI to pass.

## ❓ How to test this?

CI should be sufficient

## 📹 Demo

N/A

### CODE REVIEW

File: packages/hash/api/src/graph/system-types.ts [hunk 1]
```diff
@@ -396,17 +383,6 @@ const textEntityTypeInitializer = async (graphApi: GraphApi) => {
   })(graphApi);
 };
 
-/**
- * @todo: remove this dummy entity type once we are able to define the block data link type without it
- * @see https://app.asana.com/0/1202805690238892/1203015527055368/f
- */
-const dummyEntityTypeInitializer = async (graphApi: GraphApi) => {
-  return entityTypeInitializer({
-    ...types.entityType.dummy,
```
Comment (benwerner01) [inline packages/hash/api/src/graph/system-types.ts:405]: I think it should also be removed from [the `ontology-types.ts` file](https://github.com/hashintel/hash/blob/93a13184191c24979f7fc941deee754da7f796cb/packages/hash/shared/src/ontology-types.ts#L113-L116)

**Review by benwerner01 — CHANGES_REQUESTED** [review]
Looks good, think I spotted one remaining reference to the entity type

File: packages/hash/api/src/graph/system-types.ts (hunk 1 above)
Comment (Alfred-Mountfield) [reply packages/hash/api/src/graph/system-types.ts:405]: Thanks, kind of annoyed I missed these 😅

**Review by thehabbos007 — APPROVED** [review]
Thanks, Alfie! Looks good to me :)

**Review by benwerner01 — DISMISSED** [review]

File: packages/hash/api/src/graph/util.ts [hunk 2]
```diff
@@ -218,11 +223,10 @@ export type EntityTypeCreatorParams = {
   }[];
   outgoingLinks?: {
     linkEntityType: EntityTypeWithMetadata;
-    destinationEntityTypes: (
-      | EntityTypeWithMetadata
-      // Some models may reference themselves. This marker is used to stop infinite loops during initialization by telling the initializer to use a self reference
-      | "SELF_REFERENCE"
-    )[];
+    destinationEntityTypes?: [
```
Comment (Alfred-Mountfield) [inline packages/hash/api/src/graph/util.ts:226]: Made this optional, the Type System doesn't allow an empty `oneOf`, it allows an _optional_ `oneOf`

**Review by thehabbos007 — DISMISSED** [review]

**Review by thehabbos007 — APPROVED** [review]

**Pull request merged**